SGAPI/ItemSets/SGItemSet.m

```diff
@@ -89,7 +89,16 @@ - (void)cancelFetch {
 - (void)processResults:(NSData *)data {
     __weakSelf me = self;
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+        NSError *error = nil;
+        NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        if (error) {
+            if (me.onPageLoadFailed) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    me.onPageLoadFailed(error);
+                });
+            }
+            return;
```

Comment:
Ooh you mean that if there's an error and that property doesn't have a block, then nothing happens? In that case I'm fine with that. If the SGItemSet doesn't have a fail handler block, then it's not gonna handle fails. That's how it goes.